ts/packages/agents/oracle/src/oracleActionHandler.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+    ActionContext,
+    ActionResult,
+    ActionResultSuccess,
+    AppAction,
+    AppAgent,
+    ParameterDefinitions,
+    ParsedCommandParams,
+} from "@typeagent/agent-sdk";
+import { createActionResultFromTextDisplay } from "@typeagent/agent-sdk/helpers/action";
+import { OracleAction } from "./oracleSchema.js";
+import {
+    CommandHandler,
+    CommandHandlerTable,
+    getCommandInterface,
+} from "@typeagent/agent-sdk/helpers/command";
+
+class RequestCommandHandler implements CommandHandler {
+    public readonly description = "Send a request to the Oracle";
+    public readonly parameters: ParameterDefinitions = {
+        // Must have a single string parameter and implicit quotes
+        // in order to support '@config request <agent>'
+        args: {
+            question: {
+                description: "Request for Oracle",
+                type: "string",
+                optional: false,
+                implicitQuotes: true,
+            },
+        },
+    };
+    public async run(
+        context: ActionContext<OracleActionContext>,
+        params: ParsedCommandParams<ParameterDefinitions>,
+    ): Promise<void> {
+        if (typeof params.args?.question === "string") {
+            const result = makeRandomResponse(params.args.question);
+            context.actionIO.appendDisplay(result.displayContent);
+        }
+    }
+}
+
+const handlers: CommandHandlerTable = {
+    description: "Oracle commands",
+    defaultSubCommand: "request",
+    commands: {
+        // Command name "request" is special for '@config request <agent>'
+        request: new RequestCommandHandler(),
+    },
+};
+
+export function instantiate(): AppAgent {
+    return {
+        initializeAgentContext: initializeOracleContext,
+        executeAction: executeOracleAction,
+        ...getCommandInterface(handlers),
+    };
+}
+
+type OracleActionContext = {
+    enabled: boolean;
```

Comment:
This is a vestige of an earlier design. I would take it out but right now I can't even push to my own fork due to auth issues on my new laptop. :-(